internal/pkg/types/atto_fil_test.go

```diff
@@ -369,43 +269,43 @@ func TestString(t *testing.T) {
 
 	// A very large number of attoFIL
 	attoFIL, _ := new(big.Int).SetString("912129289198393123456789012345678", 10)
-	assert.Equal(t, "912129289198393.123456789012345678", NewAttoFIL(attoFIL).String())
+	assert.Equal(t, "912129289198393123456789012345678", NewAttoFIL(attoFIL).String())
 
 	// A multiple of 1000 attoFIL
 	attoFIL, _ = new(big.Int).SetString("9123372036854775000", 10)
-	assert.Equal(t, "9.123372036854775", NewAttoFIL(attoFIL).String())
+	assert.Equal(t, "9123372036854775000", NewAttoFIL(attoFIL).String())
 
 	// Less than 10^18 attoFIL
 	attoFIL, _ = new(big.Int).SetString("36854775878", 10)
-	assert.Equal(t, "0.000000036854775878", NewAttoFIL(attoFIL).String())
+	assert.Equal(t, "36854775878", NewAttoFIL(attoFIL).String())
 
 	// A multiple of 100 attFIL that is less than 10^18
 	attoFIL, _ = new(big.Int).SetString("36854775800", 10)
-	assert.Equal(t, "0.0000000368547758", NewAttoFIL(attoFIL).String())
+	assert.Equal(t, "36854775800", NewAttoFIL(attoFIL).String())
 
 	// A number of attFIL that is an integer number of FIL
 	attoFIL, _ = new(big.Int).SetString("123000000000000000000", 10)
-	assert.Equal(t, "123", NewAttoFIL(attoFIL).String())
+	assert.Equal(t, "123000000000000000000", NewAttoFIL(attoFIL).String())
 }
 
 func TestNewAttoFILFromFILString(t *testing.T) {
 	tf.UnitTest(t)
 
 	t.Run("parses legitimate values correctly", func(t *testing.T) {
 		attoFIL, _ := NewAttoFILFromFILString(".12345")
-		assert.Equal(t, BigIntFromString("123450000000000000"), attoFIL.val)
+		assert.Equal(t, BigIntFromString("123450000000000000"), *attoFIL.Int)
 
 		attoFIL, _ = NewAttoFILFromFILString("000000.000000")
-		assert.Equal(t, BigIntFromString("0"), attoFIL.val)
+		assert.Equal(t, BigIntFromString("0"), *attoFIL.Int)
 
 		attoFIL, _ = NewAttoFILFromFILString("0000.12345")
-		assert.Equal(t, BigIntFromString("123450000000000000"), attoFIL.val)
+		assert.Equal(t, BigIntFromString("123450000000000000"), *attoFIL.Int)
 
 		attoFIL, _ = NewAttoFILFromFILString("12345.0")
-		assert.Equal(t, BigIntFromString("12345000000000000000000"), attoFIL.val)
+		assert.Equal(t, BigIntFromString("12345000000000000000000"), *attoFIL.Int)
 
 		attoFIL, _ = NewAttoFILFromFILString("12345")
-		assert.Equal(t, BigIntFromString("12345000000000000000000"), attoFIL.val)
+		assert.Equal(t, BigIntFromString("12345000000000000000000"), *attoFIL.Int)
 	})
 
 	t.Run("rejects nonsense values", func(t *testing.T) {
```

Comment:
We shouldn't need any tests because we shouldn't have any novel functionality regarding bigints. Except possibly formatting FIL values as human-scale strings.